### PR TITLE
added open ldap end to end tests

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
Test Result 

```
(env) /home/okhatavk/Satellite/robottelo (open_ldap_tests) $ pytest tests/foreman/ -k open_ldap
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.8, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo
plugins: forked-1.1.3, xdist-1.31.0, cov-2.8.1, mock-1.10.4, services-1.3.1
collecting 2527 items / 1 skipped / 2526 selected                                                                                                                                            2020-09-28 13:22:58 - conftest - DEBUG - Collected 2666 test cases
collected 2666 items / 2663 deselected / 1 skipped / 2 selected                                                                                                                              

tests/foreman/cli/test_ldapauthsource.py .                                                                                                                                             [ 33%]
tests/foreman/ui/test_ldap_authentication.py .s                                                                                                                                        [100%]

====================================================================================== warnings summary ======================================================================================
env/lib/python3.6/site-packages/_pytest/mark/structures.py:337
  /home/okhatavk/Satellite/robottelo/env/lib/python3.6/site-packages/_pytest/mark/structures.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.ldap - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

env/lib/python3.6/site-packages/_pytest/mark/structures.py:337
  /home/okhatavk/Satellite/robottelo/env/lib/python3.6/site-packages/_pytest/mark/structures.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.high - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

env/lib/python3.6/site-packages/_pytest/mark/structures.py:337
  /home/okhatavk/Satellite/robottelo/env/lib/python3.6/site-packages/_pytest/mark/structures.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.BZ_1883209 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================================================= 2 passed, 2 skipped, 2663 deselected, 3 warnings in 267.39 seconds =============================================================

```
Git_lab PR : 358

### CODE REVIEW

File: tests/foreman/ui/test_ldap_authentication.py
Comment (mirekdlugosz) [inline tests/foreman/ui/test_ldap_authentication.py:1423]: Don't we have fixtures for that? Can't you use them?

I'm not sure if I follow here. Looks like you use this organization and location when creating ldapauthentication entity, so what would be a problem if you started logged in already in them?

File: tests/foreman/ui/test_ldap_authentication.py
Comment (mirekdlugosz) [inline tests/foreman/ui/test_ldap_authentication.py:1447]: `.split(',')[0][3:]` - could we make that less cryptic? Why user name must be split on `,`? Why are we dropping first three characters?

File: tests/foreman/ui/test_ldap_authentication.py
Comment (mirekdlugosz) [inline tests/foreman/ui/test_ldap_authentication.py:1451]: docstring doesn't talk about this segment at all. Why is it here? What are we checking?

File: tests/foreman/ui/test_ldap_authentication.py
Comment (omkarkhatavkar) [reply tests/foreman/ui/test_ldap_authentication.py:1447]: I'm, using the username from this property `username=cn=<username>,ou=users,dc=<domain>,dc=com` rather saving same username in property

File: robottelo.properties.sample
Comment (omkarkhatavkar) [reply robottelo.properties.sample:148]: good point, it as earlier used in AD and IPA. Hence I kept it like that, I too not that fan of abbreviated names

File: tests/foreman/ui/test_ldap_authentication.py
Comment (omkarkhatavkar) [reply tests/foreman/ui/test_ldap_authentication.py:1451]: Here we are checking that the user should be logged incorrectly. It should not have the permissions related to user group

File: pytest_fixtures/satellite_auth.py
Comment (omkarkhatavkar) [reply pytest_fixtures/satellite_auth.py:113]: These are dynamic properties, I really not think we should keep them in settings, as a base on our test we can change them as well. What are the benefits we can see if we put them in settings? We can end-up with multiple combinations of settings. 

Not sure can we parameterize the fixture? if yes that is a good thing with observation. In case if we want https auth source we can pass that from the test.

**Review by mirekdlugosz — APPROVED** [review]
All my comments are resolved. Thanks for update, Omkar.
ACK

**Comment by omkarkhatavkar** [comment]
We will wait till 358 git-lab PR get merged

File: pytest_fixtures/satellite_auth.py
Comment (JacobCallahan) [inline pytest_fixtures/satellite_auth.py:54]: I think this is a bit overkill to wrap the settings in a fixture, but if it is something others want to stick with, it would be better to do it like this.
```suggestion
    return settings.open_ldap
```

File: pytest_fixtures/satellite_auth.py
Comment (JacobCallahan) [inline pytest_fixtures/satellite_auth.py:114]: With the change I mentioned above (again, if needed to be in a fixture), would look like this.
```suggestion
        account=open_ldap_data.username,
        account_password=open_ldap_data.password,
        base_dn=open_ldap_data.base_dn,
        attr_firstname=LDAP_ATTR['firstname'],
        attr_lastname=LDAP_ATTR['surname'],
        attr_login=LDAP_ATTR['login'],
        server_type=LDAP_SERVER_TYPE['API']['posix'],
        attr_mail=LDAP_ATTR['mail'],
        name=gen_string('alpha'),
        host=open_ldap_data.hostname,
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

**Review by lhellebr — APPROVED** [review]
LGTM, pending Jacob.

**Comment by omkarkhatavkar** [comment]
we also need to merge the gitlab PR 358 along with this

File: tests/foreman/cli/test_ldapauthsource.py
Comment (JacobCallahan) [inline tests/foreman/cli/test_ldapauthsource.py:353]: With the change to the fixture, these should change as well. Can you also run the tests again?

File: tests/foreman/cli/test_ldapauthsource.py
Comment (JacobCallahan) [inline tests/foreman/cli/test_ldapauthsource.py:357]: same here

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_ldapauthsource.py
Comment (omkarkhatavkar) [reply tests/foreman/cli/test_ldapauthsource.py:353]: missed that thanks

**Review by JacobCallahan — APPROVED** [review]
ACK

**Pull request merged**
